### ISSUE TITLE
How the fuck do you install this?

### ISSUE THREAD

**Issue by hitgub043** [blackhole89/notekit#154]
The instructions say to "click the first entry and download the artifact", with a link to some developer forum. The first "entry" there is apparently a bug report, and there's no "artifact" anywhere to be found.

Where's the fucking .exe file?

**Comment by sp1ritCS** [comment]
If you select a successful CI build, there should be artifacts there:
![Bildschirmfoto vom 2023-07-15 10-49-03](https://github.com/blackhole89/notekit/assets/31540351/9279239d-664b-46c0-83eb-7665d56747a0)

however, the last valid artifact is expired by now. You could use the artifact from here: https://github.com/sp1ritCS/notekit/actions/runs/4990659604

**Comment by Platorius** [comment]
I'm also trying to install this. 

When i go to the page linked from sp1ritCS there is actually nothing to download. 

So what do i have to download and how do i install it for Windows ?

**Comment by YosypC** [comment]
The artifacts are all expired. Please update!

**Comment by LyesSaadi** [comment]
[I rebuilt the artifact](https://github.com/LyesSaadi/notekit/actions/runs/6520450248), you can find it here (put it in a drive so it doesn't expire again):
https://drive.proton.me/urls/8CZRASYDVM#DKtvFOQ3nFCR

**Comment by asmwarrior** [comment]
> [I rebuilt the artifact](https://github.com/LyesSaadi/notekit/actions/runs/6520450248), you can find it here (I put it in a drive so it doesn't expire again): https://drive.proton.me/urls/8CZRASYDVM#DKtvFOQ3nFCR

I have the same issue here. Thanks for sharing the Windows builds.

**Comment by asmwarrior** [comment]
One comments about this zip file. Could the exe file(notekit.exe) have an icon?

When I unzip the file, I see a lot of exe files, and all those files has the same icon(like old DOS style icon).

**Comment by asmwarrior** [comment]
Another issue I see is that:

Under Windows, I do not see the cursor in the editor. The cursor should indicates where I'm editing
Also, another issue is that I don't know how to open a external markdown file, I mean when I create a new markdown file, I see it is located under the folder `<unzip_folder>\notekit_mingw\notekit\notesbase\*.md`, but I don't see some menu item open a markdown file under Windows.
I see there is an option to set the "Base directory“ in the "preferences" dialog, but that folder option only let me to select to the root of the hard drive, I mean I could set the folder to "F:\", but not "F:\abc\".

**Comment by sp1ritCS** [comment]
> One comments about this zip file. Could the exe file(notekit.exe) have an icon?
> When I unzip the file, I see a lot of exe files, and all those files has the same icon(like old DOS style icon).

You can create a shortcut for notekit.exe and put that onto your desktop / start menu. Those shortcuts can also have custom icon, but I believe it needs to be an `.ico` file and we don't ship that.

> Under Windows, I do not see the cursor in the editor. The cursor should indicates where I'm editing

Yes, I remember there being such an issue. I think this was due to missing fonts (but not quite sure anymore); try to install the Charter font faces in `data/fonts/ ` and check if the issue still occurs.


> Also, another issue is that I don't know how to open a external markdown file, I mean when I create a new markdown file, I see it is located under the folder `<unzip_folder>\notekit_mingw\notekit\notesbase\*.md`, but I don't see some menu item open a markdown file under Windows.

NoteKit is designed to have a self contained notesbase, it's not really a editor that can edit arbitrary markdown files (mostly due to the fact that the handwriting is stored as a custom format)

> I see there is an option to set the "Base directory“ in the "preferences" dialog, but that folder option only let me to select to the root of the hard drive, I mean I could set the folder to "F:", but not "F:\abc".

Wierd. I'm not quite able to understand what your describing (you can switch been different drives but the directory structure is empty?); but try pressing `<Ctrl>L` to enter the path manually. If that doesn't work you could also set the path in `HKCU\\Software\\GSettings\\com\\github\\blackhole89\\notekit`.

**Comment by sp1ritCS** [comment]
> > 
> 
> Where can I find the font file named "Charter font"? When I looked at the `data/fonts` folder, I see there are already several files like below:
> 
> ```
> 'Charter Bold Italic.otf'  'Charter Italic.otf'   'Charter license.txt'
> 'Charter Bold.otf'         'Charter Regular.otf'
> ```

yeah, install those otf files into the system font store (user works probably too)

> OK, I understand this. So, I have already copy some `.md` file to the folder ``<unzip_folder>\notekit_mingw\notekit\notesbase`, but still I need a way to open the copied markdown file. It is really strange this software does not have a "open" file menu item.

The files and sub directories of the base directory should be displayed in the sidebar like so:
![Bildschirmfoto vom 2024-02-22 15-30-27](https://github.com/blackhole89/notekit/assets/31540351/6c92ab89-4e67-4d9c-aef6-f3927f47842f)
clicking a file should result in the file being opened. I suspect that since you're base directory is incorrectly configured the notes don't show up (but assuming that the base dir is the  F: root, it should display all directories within it 🤷🏻‍♂️ )

> I mean if I select the folder "F:\abc\def", and close the "preferences" dialog, the next time I open the dialog, it shows the "F:" drive, which means I can only set the file path as the drive root folder.

Huh, thats obviously not supposed to happen. I'm interested to know what the base dir is currently set to.  
You might also want to try running notekit with `NK_WIN32_DEBUG` set. This spawns a ConHost dialog that is attached to stdout and stderr of notekit where some sort of assertion failure is probably getting logged.

**Comment by maxpicklez** [comment]
> If you select a successful CI build, there should be artifacts there: ![Bildschirmfoto vom 2023-07-15 10-49-03](https://private-user-images.githubusercontent.com/31540351/253730504-9279239d-664b-46c0-83eb-7665d56747a0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzMzY5OTgsIm5iZiI6MTczNDMzNjY5OCwicGF0aCI6Ii8zMTU0MDM1MS8yNTM3MzA1MDQtOTI3OTIzOWQtNjY0Yi00NmMwLTgzZWItNzY2NWQ1Njc0N2EwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE2VDA4MTEzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxNjVmNDY4MGM2OTBlZTY1NDhhYmU4ZjZlNDU5YTI5NDRkZDBkZWM0NTUxYWEyYTliN2QwODc1ZjJmYzk3OTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3BpvYKEovb3dqzhk5cN2xd7QurF2bJ75GvIqVIgcNlU)
> 
> however, the last valid artifact is expired by now. You could use the artifact from here: https://github.com/sp1ritCS/notekit/actions/runs/4990659604

LOL! Anyone foolish enough to download these "artifacts" deserves the malware they have coming to them I suppose, right?

**Comment by LyesSaadi** [comment]
I mean, the logs and build process are open and are done by GitHub... By this standard, you cannot trust any linux distro and should run LFS. If you prefer, you are more than welcome to compile notekit directly.

Plus, this was built by one of the main devs of the project who has commit access to the repository. At this point, there are much easier ways to insert malware (like, in the code, for example).
